server/app/utils.py

```diff
@@ -24,4 +24,14 @@ def first(condition: Callable[[T], bool], items: Iterable[T]) -> Optional[T]:
     for item in items:
         if condition(item):
             return item
-    return None
\ No newline at end of file
+    return None
+
+def get_footer():
+    """Generates page footer. Tries to load timestamp.txt (which is generated by update.sh script)"""
+    version = "unknown"
+    try:
+        f = open("timestamp.txt", 'r')
+        version = f.read()
```

Comment:
Why you read ```timestamp.txt``` file and save its content to ```version``` variable?
If file contains timestamp then variable should be ```timestamp```. If file contains version then file should be named ```version```.

Reply:
The content of the file is just a first step. The file is expected to be recreated periodically. If there are no git commits in the mean time, the content will be the same, but the file creation and access time will be different. I expect those to be displayed as soon as I find some time to implement code that reads file timestamp.

I wanted a name that is short and descriptive and intuitively obvious what it represents. Perhaps it should be renamed to update-timestamp.txt?

Reply:
I've updated the code to what I hope is a final form of the version check. It now prints two things: SHA of the commit and a timestamp of the file modification. This tells us when the last upgrade attempt was made. In other words, if the upgrade mechanism is working or not.